From patchwork Sun Jul  1 09:05:51 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Miguel_Rodr=C3=ADguez_P=C3=A9rez?=
 <miguel@det.uvigo.gal>
X-Patchwork-Id: 10498575
Return-Path: <linux-usb-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
	[172.30.200.125])
	by pdx-korg-patchwork.web.codeaurora.org (Postfix) with ESMTP id
	5FFC1602D6 for <patchwork-linux-usb@patchwork.kernel.org>;
	Sun,  1 Jul 2018 09:06:33 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 4F1A828852
	for <patchwork-linux-usb@patchwork.kernel.org>;
	Sun,  1 Jul 2018 09:06:33 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 41A6828970; Sun,  1 Jul 2018 09:06:33 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.8 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	MAILING_LIST_MULTI, RCVD_IN_DNSWL_HI,
	T_DKIM_INVALID autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D6C3928852
	for <patchwork-linux-usb@patchwork.kernel.org>;
	Sun,  1 Jul 2018 09:06:32 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1752165AbeGAJG1 (ORCPT
	<rfc822;patchwork-linux-usb@patchwork.kernel.org>);
	Sun, 1 Jul 2018 05:06:27 -0400
Received: from mail-wr0-f193.google.com ([209.85.128.193]:42878 "EHLO
	mail-wr0-f193.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
	with ESMTP id S1752119AbeGAJGY (ORCPT
	<rfc822;linux-usb@vger.kernel.org>); Sun, 1 Jul 2018 05:06:24 -0400
Received: by mail-wr0-f193.google.com with SMTP id p1-v6so12609143wrs.9
	for <linux-usb@vger.kernel.org>; Sun, 01 Jul 2018 02:06:24 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=det.uvigo.gal; s=clouddet;
	h=from:to:cc:subject:date:message-id:in-reply-to:references
	:mime-version:content-transfer-encoding;
	bh=iyUhpxETJfFq00Mkj/nGUDk+W11ZlYXV4mPpr/pTcKQ=;
	b=GRDfKtAC7TGoGscI44Zac43sbinOdIsOoTroK15dm3EAVGyh85YCjAGU95YG4tE73x
	XlOjGRjfyJx7XH3YQoZFVyFMmGZdoCZebARPifN1Td/xrUJXixIkVaPvMU0NvAlC1hKV
	3UQJJ7laJWaNZkJk+FKAX8UTBqeZ7LzhDrrTA=
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
	d=1e100.net; s=20161025;
	h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
	:references:mime-version:content-transfer-encoding;
	bh=iyUhpxETJfFq00Mkj/nGUDk+W11ZlYXV4mPpr/pTcKQ=;
	b=PEiyUvTO/8wHUCBa1OHtfw6swdzjSdK8YJie+W9AJJ75f0d21kRRameYLUQZwHDVpq
	H30g92KVPB+ErWQcggreRJ77tVhYt2/xe43pplT+w+362yES6vICNzMle3BGtE4dl/Pa
	xwB7fmxrR1AZcHVc73EArG0UKGQvTEKA2CXyod1a2ETPjeyR0hmZA6lCiXNhCgi1fqQ0
	HDY+9MRTthPbBMhDNWV+M7x8odnp3FitYuLPcssRN9iE67HKNn/1TIm+X9QPO+QfkSD8
	ptZtoHZPFOIEh4UsuT6nBdBlzKwzFyVd9Dy87TLkPVGaIPsqC3SvtJfGSemJKXiAEloF
	P9sw==
X-Gm-Message-State: APt69E1j0cmfiozG92b8amz7HESHqGoFx4tWQbK6yoGxJ2IKRq2zmeIJ
	ut0K5C0yrx1GtX9Yn3PDnYKb3g==
X-Google-Smtp-Source: 
 AAOMgpf2vrhJjh8H5q6XcBkfR35w4R1I1GajWIrm2mkPW0sRuQ7VZw+ZXNiu9fqWW3ZrDU9dLlmjEQ==
X-Received: by 2002:adf:f888:: with SMTP id
	u8-v6mr8125377wrp.237.1530435983606;
	Sun, 01 Jul 2018 02:06:23 -0700 (PDT)
Received: from localhost.localdomain ([82.213.255.206])
	by smtp.gmail.com with ESMTPSA id
	t11-v6sm2832270wro.53.2018.07.01.02.06.22
	(version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
	Sun, 01 Jul 2018 02:06:23 -0700 (PDT)
From: =?UTF-8?q?Miguel=20Rodr=C3=ADguez=20P=C3=A9rez?=
	<miguel@det.uvigo.gal>
To: oliver@neukum.org, linux-usb@vger.kernel.org,
	netdev@vger.kernel.org, gregkh@linuxfoundation.org
Cc: =?UTF-8?q?Miguel=20Rodr=C3=ADguez=20P=C3=A9rez?= <miguel@det.uvigo.gal>
Subject: [PATCH v3 2/4] Export usbnet_cdc_update_filter
Date: Sun,  1 Jul 2018 11:05:51 +0200
Message-Id: <20180701090553.7776-3-miguel@det.uvigo.gal>
X-Mailer: git-send-email 2.17.1
In-Reply-To: <20180701090553.7776-1-miguel@det.uvigo.gal>
References: <20180701081550.GA7048@kroah.com>
	<20180701090553.7776-1-miguel@det.uvigo.gal>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Sender: linux-usb-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-usb.vger.kernel.org>
X-Mailing-List: linux-usb@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

This makes the function avaiable to other drivers, like cdn_ncm.

Signed-off-by: Miguel Rodríguez Pérez <miguel@det.uvigo.gal>
Acked-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/net/usb/cdc_ether.c | 3 ++-
 include/linux/usb/usbnet.h  | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/usb/cdc_ether.c b/drivers/net/usb/cdc_ether.c
index 815ed0dc18fe..54472ab77b90 100644
--- a/drivers/net/usb/cdc_ether.c
+++ b/drivers/net/usb/cdc_ether.c
@@ -75,7 +75,7 @@ static const u8 mbm_guid[16] = {
 	0xa6, 0x07, 0xc0, 0xff, 0xcb, 0x7e, 0x39, 0x2a,
 };
 
-static void usbnet_cdc_update_filter(struct usbnet *dev)
+void usbnet_cdc_update_filter(struct usbnet *dev)
 {
 	struct net_device *net = dev->net;
 
@@ -99,6 +99,7 @@ static void usbnet_cdc_update_filter(struct usbnet *dev)
 			NULL,
 			0);
 }
+EXPORT_SYMBOL_GPL(usbnet_cdc_update_filter);
 
 /* probes control interface, claims data interface, collects the bulk
  * endpoints, activates data interface (if needed), maybe sets MTU.
diff --git a/include/linux/usb/usbnet.h b/include/linux/usb/usbnet.h
index e2ec3582e549..7821cf1dcd60 100644
--- a/include/linux/usb/usbnet.h
+++ b/include/linux/usb/usbnet.h
@@ -286,4 +286,5 @@ extern void usbnet_update_max_qlen(struct usbnet *dev);
 extern void usbnet_get_stats64(struct net_device *dev,
 			       struct rtnl_link_stats64 *stats);
 
+extern void usbnet_cdc_update_filter(struct usbnet *);
 #endif /* __LINUX_USB_USBNET_H */
